File: JavaScript/1021 - Notas e Moedas.js
/**
 * @author Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
 * @copyright Josélio Júnior (Lunatic Fox) 2021
*/

const p = require('fs').readFileSync('/dev/stdin', 'utf8'),
l = p.split('\n');

let v = +l.shift();
const n = [100, 50, 20, 10, 5, 2];
const m = [1, .5, .25, .1, .05, .01];

console.log('NOTAS:');
n.forEach(e => {
    console.log(`${parseInt(v / e)} nota(s) de R$ ${e.toFixed(2)}`);
    v = +(v % e).toFixed(2);
});

console.log('MOEDAS:');
m.forEach(e => {
    console.log(`${parseInt(v / e)} moeda(s) de R$ ${e.toFixed(2)}`);
    v = +(v % e).toFixed(2);
});

File: Lua/1021 - Notas e Moedas.lua
--- author: Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
--- copyright: Josélio Júnior (Lunatic Fox) 2022

local v, n, m = io.read 'n',
    { 100, 50, 20, 10, 5, 2 },
    { 1, .5, .25, .1, .05, .01 }
print 'NOTAS:'
for _, e in ipairs(n) do
  print(('%d nota(s) de R$ %.2f'):format(math.floor(v / e), e))
  v = ('%.2f'):format(v % e)
end
print 'MOEDAS:'
for _, e in ipairs(m) do
  print(('%d moeda(s) de R$ %.2f'):format(math.floor(v / e), e))
  v = ('%.2f'):format(v % e)
end
